src/client/dfuse/il/int_read.c

```diff
@@ -15,20 +15,59 @@
 static ssize_t
 read_bulk(char *buff, size_t len, off_t position, struct fd_entry *entry, int *errcode)
 {
-	daos_size_t read_size = 0;
-	d_iov_t     iov       = {};
-	d_sg_list_t sgl       = {};
-	int         rc;
+	daos_size_t	read_size = 0;
+	d_iov_t		iov       = {};
+	d_sg_list_t	sgl       = {};
+	daos_event_t	ev;
+	daos_handle_t	eqh;
+	int		rc;
 
 	DFUSE_TRA_DEBUG(entry->fd_dfsoh, "%#zx-%#zx", position, position + len - 1);
 
 	sgl.sg_nr = 1;
 	d_iov_set(&iov, (void *)buff, len);
 	sgl.sg_iovs = &iov;
-	rc = dfs_read(entry->fd_cont->ioc_dfs, entry->fd_dfsoh, &sgl, position, &read_size, NULL);
+
+	rc = ioil_get_eqh(&eqh);
+	if (rc == 0) {
```

Comment:
There are several exit paths here where *errcode does not get set, I think you want a label at L69 and then replace L39-40, L48-49 and L57-58 with D_GOTOS.   The rc values at L64 and L66 will be not be propagated with this code as it is now.

Reply:
ah do you mean when the call actually succeeds that for line 64 and 66 we do not set *errcode?

Reply:
nevermind i see it